frontend/src/pages/Confirmation/ActionsTable.js

```diff
@@ -63,63 +73,88 @@ const styles = {
     fontSize: "14px",
     borderBottom: "unset",
     padding: "0px 3px 4px 8px",
+    alignSelf: "center",
     flex: 1,
     overflow: "hidden",
     textOverflow: "ellipsis",
-    '& div': {
-      width: '150px'
-    }
+    maxWidth: "140px"
+    // "& div": {
+    //   width: "100px"
+    // }
   }
 };
 
-function generateHeader(classes, status) {
+function generateHeader(classes, status, actionTableColumns) {
   return (
     <TableRow className={classes.headerRow} key={"header"}>
-      <TableCell key={"header-type"} className={classes.columnHeaderCell} style={{ flex: 2 }}>
+      <TableCell key={"header-type"} className={classes.columnHeaderCell} style={{ flex: 3 }}>
         {strings.common.type}
       </TableCell>
-      <TableCell key={"header-displayName"} className={classes.columnHeaderCell} style={{ flex: 3 }}>
-        {strings.common.name}
-      </TableCell>
-      <TableCell key={"header-permission"} className={classes.columnHeaderCell} style={{ flex: 3 }}>
-        {strings.common.permission}
-      </TableCell>
-      <TableCell key={"header-identity"} className={classes.columnHeaderCell} style={{ flex: 3 }}>
-        {strings.confirmation.user_group}
-      </TableCell>
+      {actionTableColumns.nameColumn ? (
```

Comment:
a flag for every column seems weird to me. First I would rethink the structure of the component if there is another structure that is more sufficient. If the structure must stay the same i would calculate the need of a column instead of passing it, or is there a requirement to pass them?